core/src/main/scala/org/dbpedia/extraction/mappings/TemplateMapping.scala

```diff
@@ -6,180 +6,231 @@ import org.dbpedia.extraction.transform.Quad
-                // Example case we could append to existing class are where we have to different mapped templates that one is a subclass of the other
-
-                // Condition #1
-                //  Check if the root template has been mapped to the corresponding Class of this template
-                //  If the mapping already defines a corresponding class & propery then we should create a new resource
-                val condition1_createCorrespondingProperty = correspondingClass != null &&
-                  correspondingProperty != null && pageClass.relatedClasses.contains(correspondingClass)
-
-                // Condition #2
-                // If we have more than one of the same template it means that we want to create multiple resources. See for example
-                // the pages: enwiki:Volkswagen_Golf , enwiki:List_of_Playboy_Playmates_of_2012
-                val pageTemplateSet = pageNode.getAnnotation(TemplateMapping.TEMPLATELIST_ANNOTATION).getOrElse(Seq.empty)
-                val condition2_template_exists = pageTemplateSet.contains(node.title.decoded)
-                if (!condition2_template_exists)
-                  node.setAnnotation(TemplateMapping.TEMPLATELIST_ANNOTATION, pageTemplateSet ++ Seq(node.title.decoded))
-
-                // Condition #3
-                // The current mapping is a subclass or a superclass of previous class or owl:Thing
-                val condition3_subclass = mapToClass.relatedClasses.contains(pageClass) || pageClass.relatedClasses.contains(mapToClass) || mapToClass.equals(classOwlThing) || pageClass.equals(classOwlThing)
-
-                // If all above conditions are met then use the main resource, otherwise create a new one
-                val instanceUri =
-                  if ( (!condition1_createCorrespondingProperty) && (!condition2_template_exists) && condition3_subclass ) subjectUri
-                  else generateUri(subjectUri, node)
-
-                //Add ontology instance
-                if (instanceUri == subjectUri) {
-                  createMissingTypes(graph, instanceUri, node)
-                }
-                else {
-                  createInstance(graph, instanceUri, node)
-                }
-
-                if (condition1_createCorrespondingProperty)
-                {
-                    //Connect new instance to the instance created from the root template
-                    graph += new Quad(context.language, DBpediaDatasets.OntologyPropertiesObjects, instanceUri, correspondingProperty, subjectUri, node.sourceIri)
-                }
-
-                //Extract properties
-                graph ++= mappings.flatMap(_.extract(node, instanceUri))
-            }
-        }
-        
-        graph
-    }
+  /**
+   * when extractor has a pre-phase
+   */
+
+  override def extract(node: TemplateNode, subjectUri: String): Seq[Quad] =
+  {
+    val pageNode = node.root
+    val graph = new ArrayBuffer[Quad]
+    logger.info("entity " + subjectUri)
```

Comment:
At the moment logging is a global issue in the Extraction Framework, and as I checked on my machine this line: `logger.info("entity " + subjectUri)` doesn't produce any message to the console or to any file (possibly I missed somewhere it). But anyway I also think that there is no need for this message during the extraction, so please remove it (also remove `private val logger = Logger.getLogger(TemplateMapping.getClass.getName)` in the line 30)

Reply:
resolved